lib/removeUnusedTextures.js

```diff
@@ -0,0 +1,56 @@
+'use strict';
+var Cesium = require('cesium');
+var defined = Cesium.defined;
+
+module.exports = removeUnusedTextures;
+
+function removeUnusedTextures(gltf, stats) {
+    var usedTextureIds = {};
+    var materials = gltf.materials;
+
+    // Build hash of used textures by iterating through materials
+    if (defined(materials)) {
```

Comment:
In addition to checking materials, we also needs to check techniques since a technique can provide a default value for a parameter.  In object-oriented terms, think of a technique as a class, and as a material as an instance of the class.

See https://github.com/KhronosGroup/glTF/blob/master/specification/README.md#techniqueparameters-1

In practice, it is unlikely that a technique will define the default value for a texture (well, at least for a diffuse texture), but it could happen.  Here is an example of a default color (not texture) defined in a technique:

https://github.com/KhronosGroup/glTF/blob/master/sampleModels/duck/glTF/duck.gltf#L287-L294